core/arch/opnd_shared.c

```diff
@@ -2074,10 +2079,12 @@ reg_get_bits(reg_id_t reg)
         return (byte)((reg - REG_START_16) % 8);
     if (reg >= REG_START_MMX && reg <= REG_STOP_MMX)
         return (byte)((reg - REG_START_MMX) % 8);
-    if (reg >= REG_START_XMM && reg <= REG_STOP_XMM)
-        return (byte)((reg - REG_START_XMM) % 8);
-    if (reg >= REG_START_YMM && reg <= REG_STOP_YMM)
-        return (byte)((reg - REG_START_YMM) % 8);
+    if (reg >= DR_REG_START_XMM && reg <= DR_REG_STOP_XMM)
+        return (byte)((reg - DR_REG_START_XMM) % 8);
```

Comment:
This is probably right but I don't know how 16-31 are selected -- please confirm this is correct.
Please check all the other STOP_XMM uses.

Reply:
This function is old. There was already a problem with AVX2 as there are more register bits than 3bit modrm. I extended what was already there. I'd say we augment the API with a new function later, independent from this PR?